rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderPegoutTxIndexTests.java

```diff
@@ -0,0 +1,511 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.config.BridgeConstants;
+import co.rsk.config.BridgeMainNetConstants;
+import co.rsk.core.RskAddress;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.ethereum.core.Repository;
+import org.ethereum.vm.DataWord;
+import org.ethereum.vm.PrecompiledContracts;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+
+import static co.rsk.peg.BridgeStorageIndexKey.PEGOUT_TX_SIG_HASH;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class BridgeStorageProviderPegoutTxIndexTests {
+
+    private static final String DUPLICATED_INSERTION_ERROR_MESSAGE = "Given pegout tx sigHash %s already exists in the index. Index entries are considered unique.";
+    private static final byte TRUE_VALUE = (byte) 1;
+    private final BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
+
+    private static Stream<Arguments> null_sigHash_parameters() {
+        return Stream.of(
+            Arguments.of(false),
+            Arguments.of(true)
+        );
+    }
+
+    private static Stream<Arguments> invalid_entry_values() {
+        return Stream.of(
+            Arguments.of(new byte[]{1, 2}),
+            Arguments.of(new byte[]{0})
+        );
+    }
+
+    private static Stream<Arguments> valid_sigHash_parameters() {
+        return Stream.of(
+            Arguments.of(false, Sha256Hash.ZERO_HASH),
+            Arguments.of(false, PegTestUtils.createHash(10)),
```

Comment:
Let's add this function to BitcoinTestUtils and start using it from there. We can still leave the one in PegTestUtils for the moment since it's probably used in many places, but for this class and new tests let's use the one in BitcoinTestUtils

Reply:
Done